src/System.Windows.Forms/src/System/Windows/Forms/ICommandProvider.cs

```diff
@@ -0,0 +1,155 @@
+        bool Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the previous value of the <see cref="Enabled"/> property, so
+        /// that it can be restored to its original value on assignment of a new command.
+        /// </summary>
+        protected bool? PreviousEnabledStatus { get; set; }
+
+        /// <summary>
+        /// An implementation should raise the <see cref="CommandChanged"/> event 
+        /// by calling component's or control's OnRaiseCommandChanged method.
+        /// </summary>
+        /// <param name="e">An empty <see cref="EventArgs"/> instance.</param>
+        protected void RaiseCommandChanged(EventArgs e);
+
+        /// <summary>
+        /// An implementation should raise the <see cref="CommandCanExecuteChanged"/> event 
+        /// by calling component's or control's OnCommandCanExecuteChanged method.
+        /// </summary>
+        /// <param name="sender">The sender of the event, passed from the data source.</param>
+        /// <param name="e">An empty <see cref="EventArgs"/> instance.</param>
+        protected void RaiseCommandCanExecuteChanged([AllowNull] object sender, EventArgs e);
+
+        /// <summary>
+        /// An implementation should raise the <see cref="CommandExecuting"/> event 
+        /// by calling component's or control's OnCommandExecuting method.
+        /// </summary>
+        /// <param name="e"></param>
+        protected void RaiseCommandExecuting(CancelEventArgs e);
+
+        /// <summary>
+        /// Method which should be called by a class implementing this 
+        /// interface in the setter of the <see cref="Command"/> property.
+        /// </summary>
+        /// <param name="commandComponent">Instance of the class implementing this interface.</param>
+        /// <param name="newCommand">The new value of the <see cref="Command"/>
+        /// which should be assigned to the property.</param>
+        /// <param name="commandBackingField">The backing field for
+        /// the <see cref="Command"/> property.</param>
+        protected static void CommandSetter(
+            ICommandProvider commandComponent,
+            ICommand? newCommand,
+            ref ICommand? commandBackingField)
+            => commandComponent.CommandSetter(newCommand, ref commandBackingField);
+
+        /// <summary>
+        /// Method which should be called by the class implementing this
+        /// interface, when the assigned <see cref="Command"/> should be executed.
+        /// </summary>
+        /// <remarks>
+        /// As an example, a <see cref="Button"/> should call this method inside the method which
+        /// also raises the <see cref="Control.Click"/> Event of that Button, which
+        /// would be the <see cref="Button.OnClick(EventArgs)"/> OnClick method.
+        /// </remarks>
+        /// <param name="commandComponent"></param>
+        protected static void RequestCommandExecute(ICommandProvider commandComponent)
```

Comment:
We need to validate the args on the public API, and add xml-docs for the ANE.
```suggestion
        protected static void RequestCommandExecute(ICommandProvider!! commandComponent)
```
See https://github.com/dotnet/runtime/pull/64720.

Ditto other places that allow non-nullable input args.